AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java

```diff
@@ -1065,11 +1065,7 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
     // Get the did of the parent deck (ignoring any subdecks)
     protected long getParentDid() {
         long deckID;
-        try {
-            deckID = getCol().getDecks().current().getLong("id");
-        } catch (JSONException e) {
-            throw new RuntimeException(e);
-        }
+        deckID = getCol().getDecks().selected();
```

Comment:
should look like this but without the declaration above - they should be on one line

```suggestion
        long deckID = getCol().getDecks().selected();
```

Reply:
Done